### ISSUE TITLE
 Cannot read property 'slice' of undefined

### ISSUE THREAD

**Issue by Rowleen** [andrerpena/react-mde#71]
When I try to use some button to format text, I get this warning  

> Cannot read property 'slice' of undefined

It could be because I'm passing value as string to component?

This is my ReactMde component. It receives the props from his parent.
```
<ReactMde
        {...{ value, onChange }}
        textAreaProps={{
          id,
          style: { resize: 'none' },
          value,
        }}
        commands={ReactMdeCommands.getDefaultCommands()}
        visibility={{ preview: false }}
/>
```
And this's his parent:
```
<MarkDownEditor
  id="text"
  value={this.state.reactMdeValue.text}
  onChange={this.handleChange}
/>
```
This is the error:

> Uncaught (in promise) TypeError: Cannot read property 'slice' of undefined
>     at Object.insertText (ReactMdeTextHelper.js:21)
>     at Object.makeACommandThatInsertsBeforeAndAfter (ReactMdeCommandHelper.js:55)
>     at Object.execute (ReactMdeCommands.js:34)
>     at ReactMde.<anonymous> (ReactMde.js:83)
>     at step (ReactMde.js:50)
>     at Object.next (ReactMde.js:31)
>     at ReactMde.js:25
>     at new Promise (<anonymous>)
>     at ./node_modules/react-mde/lib/js/ReactMde.js.__awaiter (ReactMde.js:21)
>     at ReactMde._this.handleCommand (ReactMde.js:77)
>     at onClick (ReactMdeToolbar.js:16)
>     at Object../node_modules/react-dom/lib/ReactErrorUtils.js.ReactErrorUtils.invokeGuardedCallback (ReactErrorUtils.js:69)
>     at executeDispatch (EventPluginUtils.js:85)
>     at Object.executeDispatchesInOrder (EventPluginUtils.js:108)
>     at executeDispatchesAndRelease (EventPluginHub.js:43)
>     at executeDispatchesAndReleaseTopLevel (EventPluginHub.js:54)
>     at Array.forEach (<anonymous>)
>     at forEachAccumulated (forEachAccumulated.js:24)
>     at Object.processEventQueue (EventPluginHub.js:254)
>     at runEventQueueInBatch (ReactEventEmitterMixin.js:17)
>     at Object.handleTopLevel [as _handleTopLevel] (ReactEventEmitterMixin.js:27)
>     at handleTopLevelImpl (ReactEventListener.js:72)
>     at ReactDefaultBatchingStrategyTransaction.perform (Transaction.js:143)
>     at Object.batchedUpdates (ReactDefaultBatchingStrategy.js:62)
>     at Object.batchedUpdates (ReactUpdates.js:97)
>     at dispatchEvent (ReactEventListener.js:147)

Thanks in advance.

**Comment by andrerpena** [comment]
Thanks for reporting. I'll take a look

**Comment by Rowleen** [comment]
Hi @andrerpena.

I continued working with React-mde and the only function that work is **code**. The others buttons throw the same warning and doesn't formmat the text.

As you see, the compoent recieve a string and not an object in the value prop. May be for that?

**Comment by andrerpena** [comment]
@Rowleen 

I have to appologize. I DID NOT ABANDON THIS PROJECT WHATSOEVER. The problem is that during the last 3 weeks I'm solely dedicated to my moving from Brazil to Germany and I have lots of bureaucracy to solve, stuff to get sorted, people to visit. 

I know react-mde still has it's inconsistencies and as soon as I settle in Berlin in the first week of February, I'll get back to it and solve. It's not just because I really like this project. It's also because I use it in other project and I **need** it to work 100%.

Thanks for your comprehension.

**Comment by Rowleen** [comment]
hi @andrerpena.

No problem. Yesterday I solved the problem. Was happen that I was using value (coming for props) on reactMde component and the textarea. So when I tried to write, textarea spawn me [Object][object].

The Fix:
Use value prop on the component and defaultValue on textarea. And why? I have on the same page a component to create notes and a component to edit the notes, so I had to use the "same" prop on both.

Thank for your answer and hope you have a good trip.
